File: 1/taskA.cs
using System;
using System.Text;

class Program
{
    static void Main()
    {
        int t = int.Parse(Console.ReadLine().Trim());
        var sb = new StringBuilder();

        for (int i = 0; i < t; i++)
        {
            var parts = Console.ReadLine().Trim().Split(' ');
            long n = long.Parse(parts[0]);
            long a = long.Parse(parts[1]);
            long b = long.Parse(parts[2]);

            long full = n / 3;
            long rem  = n % 3;

            long cost = full * Math.Min(b, 3 * a);

            if (rem == 1) cost += Math.Min(a, b);
            else if (rem == 2) cost += Math.Min(2 * a, b);

            sb.AppendLine(cost.ToString());
        }

        Console.Write(sb);
    }
}

File: 1/taskA.fs
let solve (n: int64) (a: int64) (b: int64) =
    let full = n / 3L
    let rem  = n % 3L
    let base' = full * (min b (3L * a))
    let extra =
        match rem with
        | 0L -> 0L
        | 1L -> min a b
        | _  -> min (2L * a) b
    base' + extra

[<EntryPoint>]
let main _ =
    let t = System.Console.ReadLine() |> int
    for _ in 1 .. t do
        let p = System.Console.ReadLine().Split(' ')
        printfn "%d" (solve (int64 p[0]) (int64 p[1]) (int64 p[2]))
    0